### PR TITLE
Update documentation to clarify LLVM version requirements

### LINKED ISSUE #681: Should clarify that LLVM 16 is required.
The project cannot be built against newer versions of LLVM and requires version 16. (16.0.6 to be exact.) This does not appear to be called out in the documentation.

### PR DESCRIPTION
**Pull request by bmhowe23**
Closes #681

### CODE REVIEW

**Comment by bmhowe23** [comment]
Note: I will create another PR with this cherry-picked squashed commit targeted for NVIDIA:releases/v0.4.1 once this passes review.

**Review by bettinaheim — APPROVED** [review]

File: Building.md [hunk 1]
```diff
@@ -65,10 +65,17 @@ correct path to your cuQuantum installation by setting the environment variable
...
 
-CUDA Quantum is intended to be built using the LLVM commit that the submodule is
-set to. If you need to update the LLVM commit, or if you are not using the
-pre-built version of that commit in our development container, LLVM will need to
-be built from source. To configure our build scripts, please set the LLVM
+CUDA Quantum is intended to be built using the LLVM commit that the submodule
+is set to (currently LLVM 16.0.6). If you need to update the LLVM commit, or if
+you are not using the pre-built version of that commit in our development
+container, LLVM will need to be built from source. ***Building with LLVM >=
+17.x is not currently supported*** due to the QIR specification's requirement
```
Comment (schweitzpgi) [inline Building.md:72]: We might want to just have a link to the issue number here rather than explain the QIR limitations and such in prose in the docs?

**Review by schweitzpgi — APPROVED** [review]
Looks fine to me. Let's make sure we get Bettina's input.

Just one nitpicky suggestion.

File: Building.md (hunk 1 above)
Comment (bmhowe23) [reply Building.md:72]: @schweitzpgi - I think you're talking about Issue #16. (Let me know if you disagree.) @bettinaheim - let me know when you'd prefer. I know you had previously approved this PR before Eric's comment, but if you'd rather I link to Issue #16 per Eric's suggestion, just let me know.

Comment (bettinaheim) [reply Building.md:72]: I think it is one sentence here that very clearly states that building with 17.x is not supported. The GitHub issue can get pretty technical, so I'd just keep it at that here.

**Pull request merged**